versioned/transfer/src/test/java/org/projectnessie/versioned/transfer/TestContentMigrationToPersist.java

```diff
@@ -0,0 +1,149 @@
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.versioned.BranchName;
+import org.projectnessie.versioned.GetNamedRefsParams;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.Put;
+import org.projectnessie.versioned.ReferenceInfo;
+import org.projectnessie.versioned.VersionStore;
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapter;
+import org.projectnessie.versioned.persist.inmem.InmemoryDatabaseAdapterFactory;
+import org.projectnessie.versioned.persist.inmem.InmemoryTestConnectionProviderSource;
+import org.projectnessie.versioned.persist.tests.extension.DatabaseAdapterExtension;
+import org.projectnessie.versioned.persist.tests.extension.NessieDbAdapter;
+import org.projectnessie.versioned.persist.tests.extension.NessieDbAdapterName;
+import org.projectnessie.versioned.persist.tests.extension.NessieExternalDatabase;
+import org.projectnessie.versioned.storage.common.persist.Persist;
+import org.projectnessie.versioned.storage.inmemory.InmemoryBackendFactory;
+import org.projectnessie.versioned.storage.testextension.NessieBackendName;
+import org.projectnessie.versioned.storage.testextension.NessiePersist;
+import org.projectnessie.versioned.storage.testextension.PersistExtension;
+import org.projectnessie.versioned.storage.versionstore.VersionStoreImpl;
+import org.projectnessie.versioned.transfer.files.FileExporter;
+import org.projectnessie.versioned.transfer.files.FileImporter;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.ExportMeta;
+
+@ExtendWith(SoftAssertionsExtension.class)
+@ExtendWith({PersistExtension.class, DatabaseAdapterExtension.class})
+@NessieDbAdapterName(InmemoryDatabaseAdapterFactory.NAME)
+@NessieExternalDatabase(InmemoryTestConnectionProviderSource.class)
+@NessieBackendName(InmemoryBackendFactory.NAME)
+public class TestContentMigrationToPersist {
+  @InjectSoftAssertions protected SoftAssertions soft;
+  @NessieDbAdapter protected static DatabaseAdapter databaseAdapter;
+  @NessieDbAdapter protected static VersionStore srcStore;
+
+  @NessiePersist protected static Persist persist;
+
+  @TempDir Path dir;
+
+  private VersionStore prepareTargetRepo() {
+    // Initialize repository w/o a default branch
```

Comment:
Not sure I understand - it erases and then initialized with a default branch.

Reply:
rephrased